### PR TITLE
Localization

### PR DESCRIPTION
**Pull request by xu69ux**
**Title:** Add Localization Support with useContext

**Description:**

This pull request introduces localization support to the application using the React `useContext` hook. The changes allow for dynamic language selection and provide translations for three languages: English, Russian, and Ukrainian.

Key changes include:

1. Creation of a `LanguageContext` that holds the current language and a function to change it.
2. Addition of a `translations` file that contains translations for English, Russian, and Ukrainian. This file is structured to easily add more languages or translations in the future.
3. Modification of components to use the `LanguageContext` and display text based on the current language.
4. Integration of language selection into the Yup validation schema, allowing error messages to be displayed in the selected language.

With these changes, users can now select their preferred language, and the application will display text in that language. This enhances the user experience by providing a more personalized and accessible interface.

**Note:** The tooltip logic in the login and registration forms currently uses `useRef` and is causing tooltips to display incorrectly. This will need to be addressed in a future update. The plan is to refactor the tooltip logic to use React state instead of `useRef`, which should provide a more reliable and predictable behavior.

### CODE REVIEW

File: src/contexts/translations.ts [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+  },
+  ukr: {
+    greeting: 'привіт',
+    login: 'увійти',
+    loginTitle: 'Увійти',
+    logout: 'вийти',
+    signup: 'зареєструватися',
+    signupTitle: 'Зареєструватися',
+    welcome:
+      'щоб використовувати це офігенне додаток, увійдіть за допомогою фейкової пошти',
```
Comment (GEKKO-ops) [inline src/contexts/translations.ts:75]: щоб користуватися цим офігенним додатком, увійдіть за допомогою фейкової пошти

File: src/contexts/translations.ts [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    or: 'або',
+    madeWith: 'зроблено з',
+    by: '',
+    email: 'пошта',
+    password: 'пароль',
+    username: "ім'я користувача",
+    noAccount: 'Немає аккаунта? Тоді, ',
```
Comment (GEKKO-ops) [inline src/contexts/translations.ts:82]: Відсутній аккаунт? Тоді,

File: src/contexts/translations.ts [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+    yesAccount: 'Є аккаунт? Тоді, ',
+    tooltipUsername: 'будь ласка, використовуйте будь-яке ім’я користувача',
```
Comment (GEKKO-ops) [inline src/contexts/translations.ts:84]: будь ласка, вкажіть будь-яке ім’я користувача

File: src/contexts/translations.ts [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+    tooltipEmail: 'будь ласка, використовуйте будь-яку пошту',
```
Comment (GEKKO-ops) [inline src/contexts/translations.ts:85]: будь ласка, вкажіть будь-яку пошту

File: src/contexts/translations.ts (hunk 2 above)
Comment (xu69ux) [reply src/contexts/translations.ts:82]: дякую! поправила)

File: src/contexts/translations.ts (hunk 1 above)
Comment (xu69ux) [reply src/contexts/translations.ts:75]: дякую! поправила)

**Review by dbox7 — APPROVED** [review]

**Pull request merged**